### PR TITLE
type: better type Polymer-Angular interops

### PR DESCRIPTION
**Pull request by stephanwlee**
Currently, the Polymer interop components like `tf-storage` are poorly
typed. While we try to type them better, they are not properly
typed/annotated that more advanced compiler wrongly mangles the symbol.

This change makes the interop components Klutz friendly.

### CODE REVIEW

File: tensorboard/webapp/core/effects/core_effects.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ export class CoreEffects {
   // Ngrx assumes all Effect classes have properties that inherit from the base
   // JS Object. `tf_backend` does not, so we wrap it.
   private readonly tfBackend = {
-    ref: (document.createElement('tf-backend') as any).tf_backend,
+    ref: document.createElement('tf-backend').tf_backend,
```
Comment (psybuzz) [inline tensorboard/webapp/core/effects/core_effects.ts:57]: Why is it that the other files creating one of our Polymer elements need to import `'../tb_polymer_interop_types'` but this file does not?

File: tensorboard/webapp/tb_polymer_interop_types.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+declare global {
+  interface Document {
+    createElement(name: 'tf-backend'): TfBackendElement;
+    createElement(name: 'tf-globals'): TfGlobalsElement;
+    createElement(name: 'tf-storage'): TfStorageElement;
```
Comment (psybuzz) [inline tensorboard/webapp/tb_polymer_interop_types.ts:19]: Neat pattern.

**Review by psybuzz — APPROVED** [review]
Nice to drop `any`s!

File: tensorboard/webapp/tb_polymer_interop_types.ts (hunk 2 above)
Comment (stephanwlee) [reply tensorboard/webapp/tb_polymer_interop_types.ts]: found a neater pattern :)

File: tensorboard/webapp/core/effects/core_effects.ts (hunk 1 above)
Comment (stephanwlee) [reply tensorboard/webapp/core/effects/core_effects.ts:60]: Included.

**Pull request merged**